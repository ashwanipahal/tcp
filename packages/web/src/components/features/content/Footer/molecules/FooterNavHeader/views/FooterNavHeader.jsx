import React from 'react';
import PropTypes from 'prop-types';
import { Anchor, BodyCopy, Image } from '@tcp/core/src/components/common/atoms';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import { getIconPath } from '@tcp/core/src/utils';
import styles from '../FooterNavHeader.style';

const FooterNavHeader = ({
  className,
  titleText,
  titleObj,
  ariaLabel,
  headerAsImage,
  isSubHeader,
  colNum,
}) => {
  if (!headerAsImage) {
    return (
      <BodyCopy
        className={!isSubHeader ? className : `${className} subHeader`}
        aria-label={ariaLabel}
        data-index={colNum}
        data-locator={`col_heading_${colNum}`}
        component="p"
        fontFamily="secondary"
        fontWeight="semibold"
        fontSize="fs16"
        color="text.primary"
      >
        {titleText}
      </BodyCopy>
    );
  }
  return (
    <Anchor
      to={titleObj.url}
      aria-label={titleText}
      className={`${className} img-link`}
      dataLocator={`col_heading_${colNum}`}
    >
      <Image
        tabIndex="-1"
        aria-hidden="true"
        alt={titleObj.image_alt}
        src={getIconPath(titleObj.class)}
      />
    </Anchor>
  );
};

FooterNavHeader.propTypes = {
  className: PropTypes.string.isRequired,
  ariaLabel: PropTypes.string.isRequired,
  headerAsImage: PropTypes.bool.isRequired,
  colNum: PropTypes.number.isRequired,
  titleText: PropTypes.string.isRequired,
  titleObj: PropTypes.shape({}).isRequired,
  isSubHeader: PropTypes.bool.isRequired,
};

export default withStyles(FooterNavHeader, styles);

export { FooterNavHeader as FooterNavHeaderVanilla };
