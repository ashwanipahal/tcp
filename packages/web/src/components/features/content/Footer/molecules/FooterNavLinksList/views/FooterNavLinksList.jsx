import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import Anchor from '@tcp/core/src/components/common/atoms/Anchor';
import Button from '@tcp/core/src/components/common/atoms/Button';
import { routerPush, scrollPage } from '@tcp/core/src/utils';
import styles from '../FooterNavLinksList.style';

/**
 *
 * @param {string} className set the classname for styling purpose.
 * @param {Object} listArray link list array.
 * @param {number} colnum col number.
 * @param {boolean} isLoggedIn to check if user is logged in or not.
 * @param {Object} linkConfig all the footer link action mappings
 */
const FooterNavLinksList = ({
  className,
  listArray,
  colNum,
  isLoggedIn,
  linkConfig,
  footerActionCreator,
}) => {
  const trackOrderLink = (e, handler) => {
    e.preventDefault();
    if (!isLoggedIn) footerActionCreator(handler, { state: true });
    else routerPush('/account', '/account');
  };

  const loginModalOpenClick = (e, handler) => {
    e.preventDefault();
    footerActionCreator(handler, { state: true });
  };

  const logout = (e, handler) => {
    e.preventDefault();
    footerActionCreator(handler);
  };

  const myAccountLogin = (e, handler) => {
    e.preventDefault();
    // TO-DO : When the header is fixed at top, we can remove the scroll page call.
    scrollPage();
    footerActionCreator(handler, {
      component: 'login',
      variation: 'primary',
    });
  };
  /**
   * Callback for redux action mapped to link action type
   * @callback dispatchFn
   * @param {Object} payload
   */

  /**
   * @function getOnClickAction
   * @param {string} action - link action type
   * @param {dispatchFn} dispatchFn -  redux action mapped to link action type
   */
  const getOnClickAction = (action, dispatchFn) => {
    let onClick = null;
    if (action === 'track-order') onClick = e => trackOrderLink(e, dispatchFn);
    if (action === 'favorites') onClick = e => loginModalOpenClick(e, dispatchFn);
    if (action === 'log-out') onClick = e => logout(e, dispatchFn);
    if (action === 'my-account') onClick = e => myAccountLogin(e, dispatchFn);

    return onClick;
  };

  /**
   * @function createLink to create footer links.
   * @param {object} linkItems list of all the footer links.
   * @param {number} index index number to track all the references.
   * @returns JSX of the link.
   */
  const createNavListItem = (linkItems, index) => {
    const linkAction = linkItems.action;
    const dispatchFn = linkAction ? linkConfig[linkAction] : null;
    /*
      hideLogoutMyActLink - true - if linkAction is my-account and user is logged in.
      hideLogoutMyActLink - true - if linkAction is log-out and user is  not logged in.
       hideLogoutMyActLink - false - other wise false, to show the other links.
      This condition is to satisfy the use case, to toggle between My Account and logout
      footer links.
      Use Case - When user is logged in, only Logout link should be visible to user.
                When user is not logged in, only My Account link should be visible to user.
      Problem Scenario - As the footer links are statically configured in CMS, if this condition
              is not put, user will be seeing both the links all the time.
    */
    const hideLogoutMyActLink =
      (isLoggedIn && linkAction === 'my-account') ||
      (!isLoggedIn && linkAction === 'log-out') ||
      false;
    const onClick = linkAction ? getOnClickAction(linkAction, dispatchFn) : null;

    return !hideLogoutMyActLink ? (
      <li>
        {linkAction ? (
          <Button
            type="button"
            data-locator={`col_${colNum}_link_${index}`}
            onClick={onClick}
            link
            title={linkItems.title}
          >
            {linkItems.text}
          </Button>
        ) : (
          <Anchor
            className={className}
            noLink
            to={linkItems.url}
            anchorVariation="primary"
            fontSizeVariation="large"
            dataLocator={`col_${colNum}_link_${index}`}
            target={linkItems.target}
            title={linkItems.title}
          >
            {linkItems.text}
          </Anchor>
        )}
      </li>
    ) : null;
  };

  return (
    <ul className={`${className} list`}>
      {listArray.map((linkItems, index) => createNavListItem(linkItems, index))}
    </ul>
  );
};

FooterNavLinksList.propTypes = {
  className: PropTypes.string.isRequired,
  listArray: PropTypes.shape([]).isRequired,
  colNum: PropTypes.number.isRequired,
  isLoggedIn: PropTypes.bool,
  linkConfig: PropTypes.shape({
    'track-order': PropTypes.func,
    favorites: PropTypes.func,
    'log-out': PropTypes.func,
    'my-account': PropTypes.func,
  }).isRequired,
  footerActionCreator: PropTypes.func.isRequired,
};

FooterNavLinksList.defaultProps = {
  isLoggedIn: false,
};

export default withStyles(FooterNavLinksList, styles);

export { FooterNavLinksList as FooterNavLinksListVanilla };
