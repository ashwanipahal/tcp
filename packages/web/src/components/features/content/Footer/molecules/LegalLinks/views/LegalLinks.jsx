import React from 'react';
import PropTypes from 'prop-types';
import Anchor from '@tcp/core/src/components/common/atoms/Anchor';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import { getLocator } from '@tcp/core/src/utils';
import style from '../LegalLinks.style';

const LegalLinks = ({ className, links }) => (
  <React.Fragment>
    <ul className={className}>
      {links.map((link, index) => (
        <li key={index.toString()} data-locator={`${getLocator('legal_links')}${index}`}>
          <Anchor anchorVariation="primary" to={link.url} target={link.target} title={link.title}>
            {link.text}
          </Anchor>
        </li>
      ))}
    </ul>
  </React.Fragment>
);

LegalLinks.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string,
      title: PropTypes.text,
    })
  ),
  className: PropTypes.string.isRequired,
};

LegalLinks.defaultProps = {
  links: [],
};

export { LegalLinks as LegalLinksVanilla };
export default withStyles(LegalLinks, style);
