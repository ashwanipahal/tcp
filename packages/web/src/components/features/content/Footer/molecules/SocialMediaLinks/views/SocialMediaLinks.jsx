import React from 'react';
import PropTypes from 'prop-types';
import Image from '@tcp/core/src/components/common/atoms/Image';
import Anchor from '@tcp/core/src/components/common/atoms/Anchor';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import BodyCopy from '@tcp/core/src/components/common/atoms/BodyCopy';
import ClickTracker from '@tcp/web/src/components/common/atoms/ClickTracker';
import { getIconPath, getLocator } from '@tcp/core/src/utils';
import style from '../SocialMediaLinks.style';

const SocialMediaLinks = ({ className, connectWithUsLabel, links }) => (
  <React.Fragment>
    <div className={className}>
      <BodyCopy
        className="social-media-label"
        data-locator={getLocator('label-connect-with-us')}
        component="span"
        fontFamily={['secondary', 'primary']}
        fontSize={['fs10', 'fs10', 'fs12']}
        fontWeight={['black']}
        textAlign="center"
        color="text.secondary"
      >
        {connectWithUsLabel}
      </BodyCopy>
      <div className="social-media-pallete">
        {links.map((link, index) => (
          <ClickTracker
            as={Anchor}
            to={link.url}
            target={link.target}
            clickData={{ customEvents: ['event84'] }}
          >
            <Image
              className="social-media-icon"
              data-locator={`${getLocator('social_media_links')}${index}`}
              src={getIconPath(link.class)}
              alt={link.title}
            />
          </ClickTracker>
        ))}
      </div>
    </div>
  </React.Fragment>
);

SocialMediaLinks.propTypes = {
  className: PropTypes.string.isRequired,
  connectWithUsLabel: PropTypes.string,
  links: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string,
      icon_class: PropTypes.string,
      title: PropTypes.string,
    })
  ),
};

SocialMediaLinks.defaultProps = {
  connectWithUsLabel: '',
  links: [],
};

export { SocialMediaLinks as SocialMediaLinksVanilla };
export default withStyles(SocialMediaLinks, style);
