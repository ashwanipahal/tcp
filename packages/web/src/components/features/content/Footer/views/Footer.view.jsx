import React from 'react';
import PropTypes from 'prop-types';
import { Button, Col, Row } from '@tcp/core/src/components/common/atoms';
import errorBoundary from '@tcp/core/src/components/common/hoc/withErrorBoundary/errorBoundary';
import { isTCP } from '@tcp/core/src/utils/utils';
import BodyCopy from '@tcp/core/src/components/common/atoms/BodyCopy';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import CountrySelector from '../../Header/molecules/CountrySelector';

import {
  FooterMiddleMobile,
  FooterMiddleDesktop,
  LegalLinks,
  Copyright,
  FooterTopCandidateA,
  FooterTopCandidateB,
} from '../molecules';
import { EmailSignupModal, SmsSignupModal } from '../../../../common/molecules';
import style from '../Footer.style';
import FOOTER_CONSTANTS from '../Footer.constants';

class Footer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      showFooterTopCandidateB: false,
    };
  }

  componentDidMount() {
    const { pageName } = this.props;
    // TODO: Need to change this when proper solution for A/B test come
    if (window.location.search.match('cand-b')) {
      this.setState({ showFooterTopCandidateB: true });
    } else {
      this.setState({ showFooterTopCandidateB: false });
    }

    /**
     * condition to check for homepage only using @prop pageName
     * in footer to call @function subscriptionPopUpOnPageLoad if true
     */
    if (pageName === 'homepage') {
      this.subscriptionPopUpOnPageLoad();
    }
  }

  /**
   * @function subscriptionPopUpOnPageLoad function to show
   * the email sign up modal on first time visit of the user
   * and SMS sign up modal on second visit of the user
   * using cookies for email(@cookie COOKIE_EMAIL_PERSISTENT and @cookie COOKIE_EMAIL_SESSION)
   * and sms(@cookie COOKIE_SMS_PERSISTENT) sign up modals
   */
  subscriptionPopUpOnPageLoad = () => {
    const { openEmailSignUpModal, openSmsSignUpModal } = this.props;
    const {
      COOKIE_EMAIL_PERSISTENT,
      COOKIE_EMAIL_SESSION,
      COOKIE_SMS_PERSISTENT,
      COOKIE_MAX_AGE,
      TCP_SUB_DOMAIN,
      GYMBOREE_SUB_DOMAIN,
    } = FOOTER_CONSTANTS;
    /**
     * @function domain check for the domain setting in cookie
     * using @function isTcp from utils
     */
    const domain = isTCP() ? TCP_SUB_DOMAIN : GYMBOREE_SUB_DOMAIN;

    /**
     * @function checkCookieExist function to check the existence
     * of cookie in the browser by returning @BOOLEAN
     */
    const checkCookieExist = name => document.cookie.indexOf(name) > -1;

    /**
     * condition checks for the existence of @cookie COOKIE_EMAIL_PERSISTENT
     * if false, then creates new cookie @cookie COOKIE_EMAIL_PERSISTENT and
     * COOKIE_EMAIL_SESSION and invoke @function openEmailSignUpModal
     * else checks for existence of @cookie COOKIE_EMAIL_PERSISTENT and non-existence
     * of @cookie COOKIE_SMS_PERSISTENT
     * if the above condition evaluates true then invoke
     * @function openSmsSignUpModal and set @cookie COOKIE_SMS_PERSISTENT
     */
    if (!checkCookieExist(COOKIE_EMAIL_PERSISTENT)) {
      document.cookie = `${COOKIE_EMAIL_PERSISTENT}=true; domain=${domain}; max-age=${COOKIE_MAX_AGE}`;
      document.cookie = `${COOKIE_EMAIL_SESSION}=true; domain=${domain};`;
      openEmailSignUpModal();
    } else if (
      checkCookieExist(COOKIE_EMAIL_PERSISTENT) &&
      !checkCookieExist(COOKIE_SMS_PERSISTENT)
    ) {
      document.cookie = `${COOKIE_SMS_PERSISTENT}=true; domain=${domain}; max-age=${COOKIE_MAX_AGE}`;
      openSmsSignUpModal();
    }
  };

  render() {
    const { props } = this;
    const {
      className,
      copyrightText,
      legalLinks,
      navLinks,
      referenceID,
      getUserInfoAction,
      getOrderDetailAction,
      emailSignup,
      smsSignup,
      loginModalMountedState,
      setLoginModalMountState,
      isLoggedIn,
      linkConfig,
      footerActionCreator,
    } = props;
    const { showFooterTopCandidateB } = this.state;

    return (
      <footer className={className}>
        <Row className="footer-candidate-wrapper" fullBleed>
          {showFooterTopCandidateB ? (
            <FooterTopCandidateB {...props} />
          ) : (
            <FooterTopCandidateA {...props} />
          )}
          <EmailSignupModal buttonConfig={emailSignup} />
          <SmsSignupModal buttonConfig={smsSignup} />
        </Row>
        <Row className="footer-middle mobile" fullBleed>
          <FooterMiddleMobile
            className={className}
            navLinkItems={navLinks}
            isLoggedIn={isLoggedIn}
            linkConfig={linkConfig}
            footerActionCreator={footerActionCreator}
          />
        </Row>
        <Row className="content-wrapper" fullBleed>
          <Row className="footer-middle desktop">
            <FooterMiddleDesktop
              className={className}
              navLinks={navLinks}
              loginModalMountedState={loginModalMountedState}
              setLoginModalMountState={setLoginModalMountState}
              isLoggedIn={isLoggedIn}
              linkConfig={linkConfig}
              footerActionCreator={footerActionCreator}
            />
          </Row>
        </Row>
        <div className="footer-bottom">
          <Row className="content-wrapper">
            <Row className="fullbleed-mobile">
              <Col
                className="footer-bottom__slot--1 default-offset"
                colSize={{
                  large: 3,
                  medium: 8,
                  small: 6,
                }}
              >
                <Copyright>{copyrightText}</Copyright>
                <div className="poc-hide">
                  <Button onClick={getUserInfoAction}>Get Registered User Info</Button>
                  <Button onClick={getOrderDetailAction}>Get Order detail</Button>
                </div>
              </Col>
              <Col
                className="footer-bottom__slot--2 default-offset"
                colSize={{
                  large: 7,
                  medium: 8,
                  small: 6,
                }}
              >
                <LegalLinks links={legalLinks} />
              </Col>
              <Col
                className="footer-bottom__slot--3 default-offset"
                colSize={{
                  large: 2,
                  medium: 8,
                  small: 6,
                }}
              >
                <CountrySelector showInFooter />
              </Col>
            </Row>
            <Row fullBleed>
              <Col
                colSize={{
                  large: 12,
                  medium: 8,
                  small: 6,
                }}
              >
                <BodyCopy className="reference-id" fontSize="fs32">
                  {referenceID}
                </BodyCopy>
              </Col>
            </Row>
          </Row>
        </div>
      </footer>
    );
  }
}

Footer.propTypes = {
  className: PropTypes.string.isRequired,
  copyrightText: PropTypes.string,
  navLinks: PropTypes.shape({}),
  legalLinks: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string,
      title: PropTypes.text,
    })
  ),
  referenceID: PropTypes.string,
  getUserInfoAction: PropTypes.func.isRequired,
  getOrderDetailAction: PropTypes.func.isRequired,
  openEmailSignUpModal: PropTypes.func,
  openSmsSignUpModal: PropTypes.func,
  isLoggedIn: PropTypes.bool,
  linkConfig: PropTypes.shape({
    'track-order': PropTypes.func,
    favorites: PropTypes.func,
    'log-out': PropTypes.func,
    'my-account': PropTypes.func,
  }).isRequired,
  footerActionCreator: PropTypes.func.isRequired,
  pageName: PropTypes.string,
};

Footer.defaultProps = {
  copyrightText: '',
  legalLinks: [],
  navLinks: [],
  referenceID: '',
  openEmailSignUpModal: () => {},
  openSmsSignUpModal: () => {},
  isLoggedIn: false,
  pageName: '',
};

export default withStyles(errorBoundary(Footer), style);
export { Footer as FooterVanilla };
