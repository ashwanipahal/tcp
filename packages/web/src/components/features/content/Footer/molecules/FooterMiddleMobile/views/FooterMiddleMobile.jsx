import React from 'react';
import PropTypes from 'prop-types';
import AccordionList from '@tcp/core/src/components/common/molecules/AccordionList';
import { isCanada } from '@tcp/core/src/utils';
import Col from '@tcp/core/src/components/common/atoms/Col';
import FooterNavLinksList from '../../FooterNavLinksList';

const FooterMiddleMobile = ({
  className,
  navLinkItems,
  isLoggedIn,
  linkConfig,
  footerActionCreator,
}) => {
  return (
    <Col
      colSize={{
        large: 12,
        medium: 8,
        small: 6,
      }}
      ignoreGutter={{ small: true, medium: true }}
    >
      <AccordionList className={className} accordionItems={navLinkItems}>
        {navLinkItems.map((item, index) =>
          index === 1 && isCanada() ? null : (
            <FooterNavLinksList
              insideAcccordion
              listArray={item.links}
              isLoggedIn={isLoggedIn}
              linkConfig={linkConfig}
              footerActionCreator={footerActionCreator}
            />
          )
        )}
      </AccordionList>
    </Col>
  );
};

FooterMiddleMobile.propTypes = {
  className: PropTypes.string.isRequired,
  navLinkItems: PropTypes.shape([]).isRequired,
  isLoggedIn: PropTypes.bool,
  linkConfig: PropTypes.shape({
    'track-order': PropTypes.func,
    favorites: PropTypes.func,
    'log-out': PropTypes.func,
    'my-account': PropTypes.func,
  }).isRequired,
  footerActionCreator: PropTypes.func.isRequired,
};

FooterMiddleMobile.defaultProps = {
  isLoggedIn: false,
};

export default FooterMiddleMobile;
