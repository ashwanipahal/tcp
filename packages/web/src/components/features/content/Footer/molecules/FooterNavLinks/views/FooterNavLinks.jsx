import React from 'react';
import PropTypes from 'prop-types';
import FooterNavLinksList from '../../FooterNavLinksList';
import FooterNavHeader from '../../FooterNavHeader';

const FooterNavLinks = ({
  className,
  navLinkItems,
  updateAccordionState,
  headerAsImage,
  isSubHeader,
  colNum,
}) => {
  return (
    <div className={`${className} container-nav-link`} key={navLinkItems.id} data-index={colNum}>
      <FooterNavHeader
        headerAsImage={headerAsImage}
        titleText={navLinkItems.header.text}
        titleObj={navLinkItems.header}
        updateAccordionState={updateAccordionState}
        isSubHeader={isSubHeader}
        colNum={colNum}
      />
      <FooterNavLinksList listArray={navLinkItems.links} colNum={colNum} />
    </div>
  );
};

FooterNavLinks.propTypes = {
  className: PropTypes.string.isRequired,
  navLinkItems: PropTypes.shape({}).isRequired,
  updateAccordionState: PropTypes.func.isRequired,
  headerAsImage: PropTypes.bool.isRequired,
  isSubHeader: PropTypes.bool.isRequired,
  colNum: PropTypes.number.isRequired,
};

export default FooterNavLinks;
