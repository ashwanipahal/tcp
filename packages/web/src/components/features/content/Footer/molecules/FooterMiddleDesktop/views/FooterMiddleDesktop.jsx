import React, { Fragment } from 'react';
import Col from '@tcp/core/src/components/common/atoms/Col';
import PropTypes from 'prop-types';
import FooterNavLinks from '../../FooterNavLinks';

const FooterMiddleDesktop = ({ navLinks, className }) => {
  let numberOfNavLinkCols = navLinks.length;

  const navLinkColumns = [];

  for (let i = 2; i < navLinks.length; i += 1) {
    if (navLinks[i + 1] && navLinks[i + 1].isSubHeader) {
      // For each subheader, it is going to be one col less
      // as it will adjust in the previous column bottom itself.
      // Hence, reducing the number of nav link columns.
      numberOfNavLinkCols -= 1;
      navLinkColumns.push(
        <Col
          colSize={{
            large: 2,
            medium: 8,
            small: 6,
          }}
        >
          <FooterNavLinks
            className={className}
            navLinkItems={{ header: navLinks[i].header, links: navLinks[i].links }}
            colNum={i}
          />
          <FooterNavLinks
            className={className}
            isSubHeader
            navLinkItems={{ header: navLinks[i + 1].header, links: navLinks[i + 1].links }}
            colNum={i + 1}
          />
        </Col>
      );
      i += 1;
    } else {
      navLinkColumns.push(
        <Col
          colSize={{
            large: 2,
            medium: 8,
            small: 6,
          }}
        >
          <FooterNavLinks
            className={className}
            navLinkItems={{ header: navLinks[i].header, links: navLinks[i].links }}
            colNum={i}
          />
        </Col>
      );
    }
  }

  return (
    <Fragment>
      <Col
        colSize={{
          large: 2,
          medium: 8,
          small: 6,
        }}
      >
        <FooterNavLinks
          headerAsImage
          className={className}
          navLinkItems={{ header: navLinks[0].header, links: navLinks[0].links }}
          colNum={0}
        />
      </Col>
      <Col
        colSize={{
          large: 2,
          medium: 8,
          small: 6,
        }}
      >
        <FooterNavLinks
          headerAsImage
          className={className}
          navLinkItems={{ header: navLinks[1].header, links: navLinks[1].links }}
          colNum={1}
        />
      </Col>
      {numberOfNavLinkCols <= 5 ? (
        <Col
          className="divider"
          colSize={{
            large: 1,
            medium: 8,
            small: 6,
          }}
        />
      ) : (
        ''
      )}
      {navLinkColumns}
    </Fragment>
  );
};

FooterMiddleDesktop.propTypes = {
  navLinks: PropTypes.shape([]).isRequired,
  className: PropTypes.string.isRequired,
};

export default FooterMiddleDesktop;
