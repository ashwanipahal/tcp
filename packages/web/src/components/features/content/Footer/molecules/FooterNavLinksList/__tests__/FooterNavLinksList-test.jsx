import React from 'react';
import { shallow } from 'enzyme';
import { FooterNavLinksListVanilla } from '../views/FooterNavLinksList';

describe('FooterNavLinksList component', () => {
  it('renders default footer correctly', () => {
    const props = {
      id: {},
      className: 'footer-nav-header1',
      titleText: 'Title',
      ariaLabel: 'ABCD',
      isSubHeader: false,
      headerAsImage: false,
      updateAccordionState: () => {},
      listArray: [
        {
          title: 'gift card',
          url: '',
          text: '',
          action: '',
        },
        {
          title: 'gift card',
          url: '',
          text: '',
          action: '',
        },
        {
          title: 'gift card',
          url: '',
          text: '',
          action: '',
        },
      ],
      linkConfig: {},
      colNum: 1,
    };
    const component = shallow(<FooterNavLinksListVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });
});
