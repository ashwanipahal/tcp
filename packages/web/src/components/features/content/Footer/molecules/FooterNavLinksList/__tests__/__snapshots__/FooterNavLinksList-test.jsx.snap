// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterNavLinksList component renders default footer correctly 1`] = `
<Fragment>
  <ul
    className="footer-nav-header1 list"
  >
    <li>
      <Styled(Anchor)
        anchorVariation="primary"
        className="footer-nav-header1"
        data-locator="col_1_link_0"
        fontSizeVariation="large"
        id="giftcard"
        noLink={true}
        onClick={[Function]}
        title="gift card"
        to=""
      />
    </li>
    <li>
      <Styled(Anchor)
        anchorVariation="primary"
        className="footer-nav-header1"
        data-locator="col_1_link_1"
        fontSizeVariation="large"
        id="giftcard"
        noLink={true}
        onClick={[Function]}
        title="gift card"
        to=""
      />
    </li>
    <li>
      <Styled(Anchor)
        anchorVariation="primary"
        className="footer-nav-header1"
        data-locator="col_1_link_2"
        fontSizeVariation="large"
        id="giftcard"
        noLink={true}
        onClick={[Function]}
        title="gift card"
        to=""
      />
    </li>
  </ul>
</Fragment>
`;
